src/NLPModels.jl

```diff
@@ -320,6 +367,34 @@ function jac_op!(nlp :: AbstractNLPModel, x :: AbstractVector,
                                  false, false, prod, ctprod, ctprod)
 end
 
+"""
+    J = jac_op!(nlp, rows, cols, vals, Jv, Jtv)
+
+Return the Jacobian given by `(rows, cols, vals)` as a linear operator.
+The resulting object may be used as if it were a matrix, e.g., `J * v` or `J' * v`.
+The values `Jv` and `Jtv` are used as preallocated storage for the operations.
+"""
+function jac_op!(nlp :: AbstractNLPModel, rows :: AbstractVector{<: Integer}, cols :: AbstractVector{<: Integer}, vals :: AbstractVector, Jv :: AbstractVector, Jtv :: AbstractVector)
+  prod = @closure v -> jprod!(nlp, rows, cols, vals, v, Jv)
+  ctprod = @closure v -> jtprod!(nlp, rows, cols, vals, v, Jtv)
+  return LinearOperator{Float64}(nlp.meta.ncon, nlp.meta.nvar,
+                                 false, false, prod, ctprod, ctprod)
+end
+
+"""
+    J = jac_op!(nlp, x, rows, cols, Jv, Jtv)
+
+Return the Jacobian at `x` as a linear operator.
+The resulting object may be used as if it were a matrix, e.g., `J * v` or
+`J' * v`. `(rows, cols)` should be the sparsity structure of the Jacobian.
+The values `Jv` and `Jtv` are used as preallocated storage for the operations.
+"""
+function jac_op!(nlp :: AbstractNLPModel, x :: AbstractVector, rows :: AbstractVector{<: Integer}, cols :: AbstractVector{<: Integer}, Jv :: AbstractVector, Jtv :: AbstractVector)
+  vals = jac_coord(nlp, x)
```

Comment:
Here you do call `jac_coord`.

Reply:
This one I followed the suggestion in `AmplModel`. I'm not sold either way.